### ISSUE TITLE
AttributeError: module 'cv2.dnn' has no attribute 'DNN_BACKEND_CUDA'

### ISSUE THREAD

**Issue by abdou31** [opencv/opencv#17582]
##### Detailed description

Hello, I'm trying to execute the tutorial of Social distancing of Adrian Rosebrock .
+ https://www.pyimagesearch.com/2020/06/01/opencv-social-distancing-detector/

I have changed from `GPU=False` to  `GPU=True` in social_destancing_config.py file before executing the script, I'm got this error:
>   File "social_distance_detector.py", line 41, in <module>
    net.setPreferableBackend(cv2.dnn.DNN_BACKEND_CUDA)
AttributeError: module 'cv2.dnn' has no attribute 'DNN_BACKEND_CUDA'

How can I fix this error?

**Comment by YashasSamaga** [comment]
You need OpenCV 4.2 or above to use `DNN_BACKEND_CUDA`.

**Comment by abdou31** [comment]
I'm installing OpenCV from source and I have built it using CMake with opencv_contrib (extra modules ) and I added to the path OPENCV_dir, also I have installed opencv_python 4.2.0.  but when I tried to execute the script, I got this error:
> DNN module was not built with CUDA backend; switching to CPU

**Comment by YashasSamaga** [comment]
@abdou31 https://www.pyimagesearch.com/2020/02/03/how-to-use-opencvs-dnn-module-with-nvidia-gpus-cuda-and-cudnn/

**Comment by abdou31** [comment]
I don't have `with_cudnn` parameter

**Comment by abdou31** [comment]
I'm using, for now, CMAKE-GUI, I'm repeating right now the task of configuration and generation, I have deleted all files presented in build folder. 
![image](https://user-images.githubusercontent.com/19480228/85156971-1b6b3400-b25b-11ea-9162-873358dbb4e6.png)

the output of Cmake configuration ( some parameters are not set because the setting of parameters could be only after the configuration task on Cmake-GUI ( I can't access to values of parameters before run the configuring task).

```Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.17763.
The CXX compiler identification is MSVC 19.25.28610.4
The C compiler identification is MSVC 19.25.28610.4
Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe
Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe -- works
Detecting CXX compiler ABI info
Detecting CXX compiler ABI info - done
Detecting CXX compile features
Detecting CXX compile features - done
Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe
Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe -- works
Detecting C compiler ABI info
Detecting C compiler ABI info - done
Detecting C compile features
Detecting C compile features - done
Detected processor: AMD64
Performing Test HAVE_CXX11 (check file: cmake/checks/cxx11.cpp)
Performing Test HAVE_CXX11 - Success
Found PythonInterp: C:/ProgramData/Anaconda2/python.exe (found suitable version "2.7.15", minimum required is "2.7") 
Found PythonLibs: C:/ProgramData/Anaconda2/libs/python27.lib (found suitable exact version "2.7.15") 
Found PythonInterp: C:/Users/starinfo/AppData/Local/Programs/Python/Python36/python3.exe (found suitable version "3.6.8", minimum required is "3.2") 
Found PythonLibs: C:/Users/starinfo/AppData/Local/Programs/Python/Python36/libs/python36.lib (found suitable exact version "3.6.8") 
Performing Test HAVE_CXX_FP:PRECISE
Performing Test HAVE_CXX_FP:PRECISE - Success
Performing Test HAVE_C_FP:PRECISE
Performing Test HAVE_C_FP:PRECISE - Success
Performing Test HAVE_CPU_SSE3_SUPPORT (check file: cmake/checks/cpu_sse3.cpp)
Performing Test HAVE_CPU_SSE3_SUPPORT - Success
Performing Test HAVE_CPU_SSSE3_SUPPORT (check file: cmake/checks/cpu_ssse3.cpp)
Performing Test HAVE_CPU_SSSE3_SUPPORT - Success
Performing Test HAVE_CPU_SSE4_1_SUPPORT (check file: cmake/checks/cpu_sse41.cpp)
Performing Test HAVE_CPU_SSE4_1_SUPPORT - Success
Performing Test HAVE_CPU_POPCNT_SUPPORT (check file: cmake/checks/cpu_popcnt.cpp)
Performing Test HAVE_CPU_POPCNT_SUPPORT - Success
Performing Test HAVE_CPU_SSE4_2_SUPPORT (check file: cmake/checks/cpu_sse42.cpp)
Performing Test HAVE_CPU_SSE4_2_SUPPORT - Success
Performing Test HAVE_CXX_ARCH:AVX (check file: cmake/checks/cpu_fp16.cpp)
Performing Test HAVE_CXX_ARCH:AVX - Success
Performing Test HAVE_CXX_ARCH:AVX2 (check file: cmake/checks/cpu_avx2.cpp)
Performing Test HAVE_CXX_ARCH:AVX2 - Success
Performing Test HAVE_CXX_ARCH:AVX512 (check file: cmake/checks/cpu_avx512.cpp)
Performing Test HAVE_CXX_ARCH:AVX512 - Success
Performing Test HAVE_CPU_BASELINE_FLAGS
Performing Test HAVE_CPU_BASELINE_FLAGS - Success
Performing Test HAVE_CPU_DISPATCH_FLAGS_SSE4_1
Performing Test HAVE_CPU_DISPATCH_FLAGS_SSE4_1 - Success
Performing Test HAVE_CPU_DISPATCH_FLAGS_SSE4_2
Performing Test HAVE_CPU_DISPATCH_FLAGS_SSE4_2 - Success
Performing Test HAVE_CPU_DISPATCH_FLAGS_FP16
Performing Test HAVE_CPU_DISPATCH_FLAGS_FP16 - Success
Performing Test HAVE_CPU_DISPATCH_FLAGS_AVX
Performing Test HAVE_CPU_DISPATCH_FLAGS_AVX - Success
Performing Test HAVE_CPU_DISPATCH_FLAGS_AVX2
Performing Test HAVE_CPU_DISPATCH_FLAGS_AVX2 - Success
Performing Test HAVE_CPU_DISPATCH_FLAGS_AVX512_SKX
Performing Test HAVE_CPU_DISPATCH_FLAGS_AVX512_SKX - Success
Check if the system is big endian
Searching 16 bit integer
Looking for sys/types.h
Looking for sys/types.h - found
Looking for stdint.h
Looking for stdint.h - found
Looking for stddef.h
Looking for stddef.h - found
Check size of unsigned short
Check size of unsigned short - done
Using unsigned short
Check if the system is big endian - little endian
Looking for fseeko
Looking for fseeko - not found
Check size of off64_t
Check size of off64_t - failed
libjpeg-turbo: VERSION = 2.0.4, BUILD = opencv-4.4.0-pre-libjpeg-turbo
Check size of size_t
Check size of size_t - done
Check size of unsigned long
Check size of unsigned long - done
Looking for include file intrin.h
Looking for include file intrin.h - found
Looking for assert.h
Looking for assert.h - found
Looking for fcntl.h
Looking for fcntl.h - found
Looking for inttypes.h
Looking for inttypes.h - found
Looking for io.h
Looking for io.h - found
Looking for limits.h
Looking for limits.h - found
Looking for malloc.h
Looking for malloc.h - found
Looking for memory.h
Looking for memory.h - found
Looking for search.h
Looking for search.h - found
Looking for string.h
Looking for string.h - found
Performing Test C_HAS_inline
Performing Test C_HAS_inline - Success
Check size of signed short
Check size of signed short - done
Check size of unsigned short
Check size of unsigned short - done
Check size of signed int
Check size of signed int - done
Check size of unsigned int
Check size of unsigned int - done
Check size of signed long
Check size of signed long - done
Check size of signed long long
Check size of signed long long - done
Check size of unsigned long long
Check size of unsigned long long - done
Check size of unsigned char *
Check size of unsigned char * - done
Check size of ptrdiff_t
Check size of ptrdiff_t - done
Looking for memmove
Looking for memmove - not found
Looking for setmode
Looking for setmode - found
Looking for strcasecmp
Looking for strcasecmp - not found
Looking for strchr
Looking for strchr - found
Looking for strrchr
Looking for strrchr - found
Looking for strstr
Looking for strstr - found
Looking for strtol
Looking for strtol - found
Looking for strtol
Looking for strtol - found
Looking for strtoull
Looking for strtoull - found
Looking for lfind
Looking for lfind - found
Performing Test HAVE_SNPRINTF
Performing Test HAVE_SNPRINTF - Success
Check if the system is big endian
Searching 16 bit integer
Using unsigned short
Check if the system is big endian - little endian
Performing Test HAVE_C_STD_C99
Performing Test HAVE_C_STD_C99 - Failed
Could NOT find OpenJPEG (minimal suitable version: 2.0, recommended version >= 2.3.1)
IPPICV: Download: ippicv_2020_win_intel64_20191018_general.zip
found Intel IPP (ICV version): 2020.0.0 [2020.0.0 Gold]
at: D:/opencv_4/build/3rdparty/ippicv/ippicv_win/icv
found Intel IPP Integration Wrappers sources: 2020.0.0
at: D:/opencv_4/build/3rdparty/ippicv/ippicv_win/iw
Could not find OpenBLAS include. Turning OpenBLAS_FOUND off
Could not find OpenBLAS lib. Turning OpenBLAS_FOUND off
Looking for sgemm_
Looking for sgemm_ - not found
Looking for pthread.h
Looking for pthread.h - not found
Found Threads: TRUE  
Could NOT find BLAS (missing: BLAS_LIBRARIES) 
LAPACK requires BLAS
A library with LAPACK API not found. Please specify library location.
Found JNI: C:/Program Files/Java/jdk1.8.0_211/lib/jawt.lib  
VTK is not found. Please set -DVTK_DIR in CMake to VTK build directory, or to VTK install subdirectory with VTKConfig.cmake file
ADE: Download: v0.1.1f.zip
OpenCV Python: during development append to PYTHONPATH: D:/opencv_4/build/python_loader
FFMPEG: Download: opencv_videoio_ffmpeg.dll
FFMPEG: Download: opencv_videoio_ffmpeg_64.dll
FFMPEG: Download: ffmpeg_version.cmake
Looking for mfapi.h
Looking for mfapi.h - found
Looking for d3d11_4.h
Looking for d3d11_4.h - found
Allocator metrics storage type: 'long long'
Registering hook 'INIT_MODULE_SOURCES_opencv_dnn': D:/opencv_4/sources/modules/dnn/cmake/hooks/INIT_MODULE_SOURCES_opencv_dnn.cmake
opencv_dnn: filter out cuda4dnn source code
CMake Warning at cmake/OpenCVGenSetupVars.cmake:54 (message):
  CONFIGURATION IS NOT SUPPORTED: validate setupvars script in install
  directory
Call Stack (most recent call first):
  CMakeLists.txt:968 (include)



General configuration for OpenCV 4.4.0-pre =====================================
  Version control:               4.3.0-434-g9ba5581d17-dirty

  Platform:
    Timestamp:                   2020-06-19T16:21:41Z
    Host:                        Windows 10.0.17763 AMD64
    CMake:                       3.14.2
    CMake generator:             Visual Studio 16 2019
    CMake build tool:            C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/MSBuild/Current/Bin/MSBuild.exe
    MSVC:                        1925

  CPU/HW features:
    Baseline:                    SSE SSE2 SSE3
      requested:                 SSE3
    Dispatched code generation:  SSE4_1 SSE4_2 FP16 AVX AVX2 AVX512_SKX
      requested:                 SSE4_1 SSE4_2 AVX FP16 AVX2 AVX512_SKX
      SSE4_1 (17 files):         + SSSE3 SSE4_1
      SSE4_2 (2 files):          + SSSE3 SSE4_1 POPCNT SSE4_2
      FP16 (1 files):            + SSSE3 SSE4_1 POPCNT SSE4_2 FP16 AVX
      AVX (5 files):             + SSSE3 SSE4_1 POPCNT SSE4_2 AVX
      AVX2 (31 files):           + SSSE3 SSE4_1 POPCNT SSE4_2 FP16 FMA3 AVX AVX2
      AVX512_SKX (7 files):      + SSSE3 SSE4_1 POPCNT SSE4_2 FP16 FMA3 AVX AVX2 AVX_512F AVX512_COMMON AVX512_SKX

  C/C++:
    Built as dynamic libs?:      YES
    C++ standard:                11
    C++ Compiler:                C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe  (ver 19.25.28610.4)
    C++ flags (Release):         /DWIN32 /D_WINDOWS /W4 /GR  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:precise     /EHa /wd4127 /wd4251 /wd4324 /wd4275 /wd4512 /wd4589 /MP  /MD /O2 /Ob2 /DNDEBUG 
    C++ flags (Debug):           /DWIN32 /D_WINDOWS /W4 /GR  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:precise     /EHa /wd4127 /wd4251 /wd4324 /wd4275 /wd4512 /wd4589 /MP  /MDd /Zi /Ob0 /Od /RTC1 
    C Compiler:                  C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe
    C flags (Release):           /DWIN32 /D_WINDOWS /W3  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:precise     /MP   /MD /O2 /Ob2 /DNDEBUG 
    C flags (Debug):             /DWIN32 /D_WINDOWS /W3  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:precise     /MP /MDd /Zi /Ob0 /Od /RTC1 
    Linker flags (Release):      /machine:x64  /INCREMENTAL:NO 
    Linker flags (Debug):        /machine:x64  /debug /INCREMENTAL 
    ccache:                      NO
    Precompiled headers:         YES
    Extra dependencies:
    3rdparty dependencies:

  OpenCV modules:
    To be built:                 calib3d core dnn features2d flann gapi highgui imgcodecs imgproc ml objdetect photo python2 python3 stitching ts video videoio
    Disabled:                    world
    Disabled by dependency:      -
    Unavailable:                 java js
    Applications:                tests perf_tests apps
    Documentation:               NO
    Non-free algorithms:         NO

  Windows RT support:            NO

  GUI: 
    Win32 UI:                    YES
    VTK support:                 NO

  Media I/O: 
    ZLib:                        build (ver 1.2.11)
    JPEG:                        build-libjpeg-turbo (ver 2.0.4-62)
    WEBP:                        build (ver encoder: 0x020f)
    PNG:                         build (ver 1.6.37)
    TIFF:                        build (ver 42 - 4.0.10)
    JPEG 2000:                   build Jasper (ver 1.900.1)
    OpenEXR:                     build (ver 2.3.0)
    HDR:                         YES
    SUNRASTER:                   YES
    PXM:                         YES
    PFM:                         YES

  Video I/O:
    DC1394:                      NO
    FFMPEG:                      YES (prebuilt binaries)
      avcodec:                   YES (58.54.100)
      avformat:                  YES (58.29.100)
      avutil:                    YES (56.31.100)
      swscale:                   YES (5.5.100)
      avresample:                YES (4.0.0)
    GStreamer:                   NO
    DirectShow:                  YES
    Media Foundation:            YES
      DXVA:                      YES

  Parallel framework:            Concurrency

  Trace:                         YES (with Intel ITT)

  Other third-party libraries:
    Intel IPP:                   2020.0.0 Gold [2020.0.0]
           at:                   D:/opencv_4/build/3rdparty/ippicv/ippicv_win/icv
    Intel IPP IW:                sources (2020.0.0)
              at:                D:/opencv_4/build/3rdparty/ippicv/ippicv_win/iw
    Lapack:                      NO
    Eigen:                       NO
    Custom HAL:                  NO
    Protobuf:                    build (3.5.1)

  OpenCL:                        YES (NVD3D11)
    Include path:                D:/opencv_4/sources/3rdparty/include/opencl/1.2
    Link libraries:              Dynamic load

  Python 2:
    Interpreter:                 C:/ProgramData/Anaconda2/python.exe (ver 2.7.15)
    Libraries:                   C:/ProgramData/Anaconda2/libs/python27.lib (ver 2.7.15)
    numpy:                       C:/ProgramData/Anaconda2/lib/site-packages/numpy/core/include (ver 1.15.4)
    install path:                C:/ProgramData/Anaconda2/Lib/site-packages/cv2/python-2.7

  Python 3:
    Interpreter:                 C:/Users/starinfo/AppData/Local/Programs/Python/Python36/python3.exe (ver 3.6.8)
    Libraries:                   C:/Users/starinfo/AppData/Local/Programs/Python/Python36/libs/python36.lib (ver 3.6.8)
    numpy:                       C:/Users/starinfo/AppData/Roaming/Python/Python36/site-packages/numpy/core/include (ver 1.16.3)
    install path:                C:/Users/starinfo/AppData/Local/Programs/Python/Python36/Lib/site-packages/cv2/python-3.6

  Python (for build):            C:/ProgramData/Anaconda2/python.exe

  Java:                          
    ant:                         NO
    JNI:                         C:/Program Files/Java/jdk1.8.0_211/include C:/Program Files/Java/jdk1.8.0_211/include/win32 C:/Program Files/Java/jdk1.8.0_211/include
    Java wrappers:               NO
    Java tests:                  NO

  Install to:                    D:/opencv_4/build/install
-----------------------------------------------------------------

Configuring done```

**Comment by abdou31** [comment]
As you can see, there is no `with_cudnn`
![image](https://user-images.githubusercontent.com/19480228/85157705-26729400-b25c-11ea-9371-51e42f0e045d.png)

**Comment by YashasSamaga** [comment]
@abdou31 
1. Set `WITH_CUDA` and `OPENCV_EXTRA_MODULES_PATH` to `path/to/opencv_contrib/modules`
2. `WITH_CUDNN` option should now appear. Set `WITH_CUDNN` and `OPENCV_DNN_CUDA`

**Comment by abdou31** [comment]
Yes I'm setting `OPENCV_EXTRA_MODULES_PATH` to the path of opencv_contrib in the first attempt.
I can't find `WITH_CUDNN`

**Comment by YashasSamaga** [comment]
@abdou31 You need to set `WITH_CUDA` also for `WITH_CUDNN` to appear.

**Comment by abdou31** [comment]
Ok, I'm setting `with_cuda` and any parameter of cuda , I'm setting also OPENCV_EXTRA_MODULES_PATH, and I'm executing the generate task

**Comment by abdou31** [comment]
I'm getting an error : 
> Could NOT find CUDNN: Found unsuitable version "7.4.2", but required is at least "7.5" (found C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0/lib/x64/cudnn.lib)
CUDA detected: 10.0
CMake Error at cmake/OpenCVDetectCUDA.cmake:111 (string):
  string sub-command STRIP requires two arguments.
Call Stack (most recent call first):
  cmake/OpenCVDetectCUDA.cmake:164 (ocv_filter_available_architecture)
  cmake/OpenCVFindLibsPerf.cmake:43 (include)
  CMakeLists.txt:688 (include)

**Comment by YashasSamaga** [comment]
@abdou31 [Download and install the latest cuDNN](https://developer.nvidia.com/cudnn). You need cuDNN 7.5 or above.

**Comment by abdou31** [comment]
I input the build and the sources folder on directory parameters of CMake-GUI and I made an another configure task after configuration is done, I have set `with_cuda`, `opencv_extra_path` and `OPENCV_DNN_CUDA` parameters and make a configure again, I'm getting this error:

```
found Intel IPP Integration Wrappers sources: 2020.0.0
at: D:/opencv_4/build/3rdparty/ippicv/ippicv_win/iw
Found CUDNN: C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0/lib/x64/cudnn.lib (found suitable version "7.5.0", minimum required is "7.5") 
CUDA detected: 10.0
CMake Error at cmake/OpenCVDetectCUDA.cmake:111 (string):
  string sub-command STRIP requires two arguments.
Call Stack (most recent call first):
  cmake/OpenCVDetectCUDA.cmake:164 (ocv_filter_available_architecture)
  cmake/OpenCVFindLibsPerf.cmake:43 (include)
  CMakeLists.txt:688 (include)


CMake Error at cmake/OpenCVDetectCUDA.cmake:125 (string):
  string sub-command REPLACE requires at least four arguments.
Call Stack (most recent call first):
  cmake/OpenCVDetectCUDA.cmake:174 (ocv_wipeout_deprecated)
  cmake/OpenCVFindLibsPerf.cmake:43 (include)
  CMakeLists.txt:688 (include)


CUDA NVCC target flags: -D_FORCE_INLINES
Could not find OpenBLAS include. Turning OpenBLAS_FOUND off
Could not find OpenBLAS lib. Turning OpenBLAS_FOUND off
Could NOT find BLAS (missing: BLAS_LIBRARIES) 
LAPACK requires BLAS
A library with LAPACK API not found. Please specify library location.
VTK is not found. Please set -DVTK_DIR in CMake to VTK build directory, or to VTK install subdirectory with VTKConfig.cmake file
OpenCV Python: during development append to PYTHONPATH: D:/opencv_4/build/python_loader
Module opencv_alphamat disabled because the following dependencies are not found: Eigen
Caffe:   NO
Protobuf:   NO
Glog:   NO
freetype2:   NO
harfbuzz:    NO
Julia not found. Not compiling Julia Bindings. 
Module opencv_ovis disabled because OGRE3D was not found
No preference for use of exported gflags CMake configuration set, and no hints for include/library directories provided. Defaulting to preferring an installed/exported gflags CMake configuration if available.
Failed to find installed gflags CMake configuration, searching for gflags build directories exported with CMake.
Failed to find gflags - Failed to find an installed/exported CMake configuration for gflags, will perform search for installed gflags components.
Failed to find gflags - Could not find gflags include directory, set GFLAGS_INCLUDE_DIR to directory containing gflags/gflags.h
Failed to find glog - Could not find glog include directory, set GLOG_INCLUDE_DIR to directory containing glog/logging.h
Module opencv_sfm disabled because the following dependencies are not found: Eigen Glog/Gflags
Tesseract:   NO
Allocator metrics storage type: 'long long'
Registering hook 'INIT_MODULE_SOURCES_opencv_dnn': D:/opencv_4/sources/modules/dnn/cmake/hooks/INIT_MODULE_SOURCES_opencv_dnn.cmake
xfeatures2d/boostdesc: Download: boostdesc_bgm.i
xfeatures2d/boostdesc: Download: boostdesc_bgm_bi.i
xfeatures2d/boostdesc: Download: boostdesc_bgm_hd.i
xfeatures2d/boostdesc: Download: boostdesc_binboost_064.i
xfeatures2d/boostdesc: Download: boostdesc_binboost_128.i
xfeatures2d/boostdesc: Download: boostdesc_binboost_256.i
xfeatures2d/boostdesc: Download: boostdesc_lbgm.i
xfeatures2d/vgg: Download: vgg_generated_48.i
xfeatures2d/vgg: Download: vgg_generated_64.i
xfeatures2d/vgg: Download: vgg_generated_80.i
xfeatures2d/vgg: Download: vgg_generated_120.i
data: Download: face_landmark_model.dat
NVIDIA_OPTICAL_FLOW: Download: 79c6cee80a2df9a196f20afd6b598a9810964c32.zip
CMake Warning at cmake/OpenCVGenSetupVars.cmake:54 (message):
  CONFIGURATION IS NOT SUPPORTED: validate setupvars script in install
  directory
Call Stack (most recent call first):
  CMakeLists.txt:968 (include)



General configuration for OpenCV 4.4.0-pre =====================================
  Version control:               4.3.0-434-g9ba5581d17-dirty

  Extra modules:
    Location (extra):            D:/opencv_contrib/modules
    Version control (extra):     4.3.0-63-g522a062c

  Platform:
    Timestamp:                   2020-06-19T17:13:52Z
    Host:                        Windows 10.0.17763 AMD64
    CMake:                       3.14.2
    CMake generator:             Visual Studio 16 2019
    CMake build tool:            C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/MSBuild/Current/Bin/MSBuild.exe
    MSVC:                        1925

  CPU/HW features:
    Baseline:                    SSE SSE2 SSE3
      requested:                 SSE3
    Dispatched code generation:  SSE4_1 SSE4_2 FP16 AVX AVX2 AVX512_SKX
      requested:                 SSE4_1 SSE4_2 AVX FP16 AVX2 AVX512_SKX
      SSE4_1 (17 files):         + SSSE3 SSE4_1
      SSE4_2 (2 files):          + SSSE3 SSE4_1 POPCNT SSE4_2
      FP16 (1 files):            + SSSE3 SSE4_1 POPCNT SSE4_2 FP16 AVX
      AVX (5 files):             + SSSE3 SSE4_1 POPCNT SSE4_2 AVX
      AVX2 (31 files):           + SSSE3 SSE4_1 POPCNT SSE4_2 FP16 FMA3 AVX AVX2
      AVX512_SKX (7 files):      + SSSE3 SSE4_1 POPCNT SSE4_2 FP16 FMA3 AVX AVX2 AVX_512F AVX512_COMMON AVX512_SKX

  C/C++:
    Built as dynamic libs?:      YES
    C++ standard:                11
    C++ Compiler:                C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe  (ver 19.25.28610.4)
    C++ flags (Release):         /DWIN32 /D_WINDOWS /W4 /GR  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:precise     /EHa /wd4127 /wd4251 /wd4324 /wd4275 /wd4512 /wd4589 /MP  /MD /O2 /Ob2 /DNDEBUG 
    C++ flags (Debug):           /DWIN32 /D_WINDOWS /W4 /GR  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:precise     /EHa /wd4127 /wd4251 /wd4324 /wd4275 /wd4512 /wd4589 /MP  /MDd /Zi /Ob0 /Od /RTC1 
    C Compiler:                  C:/Program Files (x86)/Microsoft Visual Studio/2019/Professional/VC/Tools/MSVC/14.25.28610/bin/Hostx64/x64/cl.exe
    C flags (Release):           /DWIN32 /D_WINDOWS /W3  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:precise     /MP   /MD /O2 /Ob2 /DNDEBUG 
    C flags (Debug):             /DWIN32 /D_WINDOWS /W3  /D _CRT_SECURE_NO_DEPRECATE /D _CRT_NONSTDC_NO_DEPRECATE /D _SCL_SECURE_NO_WARNINGS /Gy /bigobj /Oi  /fp:precise     /MP /MDd /Zi /Ob0 /Od /RTC1 
    Linker flags (Release):      /machine:x64  /INCREMENTAL:NO 
    Linker flags (Debug):        /machine:x64  /debug /INCREMENTAL 
    ccache:                      NO
    Precompiled headers:         YES
    Extra dependencies:          cudart_static.lib nppc.lib nppial.lib nppicc.lib nppicom.lib nppidei.lib nppif.lib nppig.lib nppim.lib nppist.lib nppisu.lib nppitc.lib npps.lib cublas.lib cudnn.lib cufft.lib -LIBPATH:C:/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v10.0/lib/x64
    3rdparty dependencies:

  OpenCV modules:
    To be built:                 aruco bgsegm bioinspired calib3d ccalib core cudaarithm cudabgsegm cudacodec cudafeatures2d cudafilters cudaimgproc cudalegacy cudaobjdetect cudaoptflow cudastereo cudawarping cudev datasets dnn dnn_objdetect dnn_superres dpm face features2d flann fuzzy gapi hfs highgui img_hash imgcodecs imgproc intensity_transform line_descriptor ml objdetect optflow phase_unwrapping photo plot python2 python3 quality rapid reg rgbd saliency shape stereo stitching structured_light superres surface_matching text tracking ts video videoio videostab xfeatures2d ximgproc xobjdetect xphoto
    Disabled:                    world
    Disabled by dependency:      -
    Unavailable:                 alphamat cnn_3dobj cvv freetype hdf java js julia matlab ovis sfm viz
    Applications:                tests perf_tests apps
    Documentation:               NO
    Non-free algorithms:         NO

  Windows RT support:            NO

  GUI: 
    Win32 UI:                    YES
    VTK support:                 NO

  Media I/O: 
    ZLib:                        build (ver 1.2.11)
    JPEG:                        build-libjpeg-turbo (ver 2.0.4-62)
    WEBP:                        build (ver encoder: 0x020f)
    PNG:                         build (ver 1.6.37)
    TIFF:                        build (ver 42 - 4.0.10)
    JPEG 2000:                   build Jasper (ver 1.900.1)
    OpenEXR:                     build (ver 2.3.0)
    HDR:                         YES
    SUNRASTER:                   YES
    PXM:                         YES
    PFM:                         YES

  Video I/O:
    DC1394:                      NO
    FFMPEG:                      YES (prebuilt binaries)
      avcodec:                   YES (58.54.100)
      avformat:                  YES (58.29.100)
      avutil:                    YES (56.31.100)
      swscale:                   YES (5.5.100)
      avresample:                YES (4.0.0)
    GStreamer:                   NO
    DirectShow:                  YES
    Media Foundation:            YES
      DXVA:                      YES

  Parallel framework:            Concurrency

  Trace:                         YES (with Intel ITT)

  Other third-party libraries:
    Intel IPP:                   2020.0.0 Gold [2020.0.0]
           at:                   D:/opencv_4/build/3rdparty/ippicv/ippicv_win/icv
    Intel IPP IW:                sources (2020.0.0)
              at:                D:/opencv_4/build/3rdparty/ippicv/ippicv_win/iw
    Lapack:                      NO
    Eigen:                       NO
    Custom HAL:                  NO
    Protobuf:                    build (3.5.1)

  NVIDIA CUDA:                   YES (ver 10.0, CUFFT CUBLAS)
    NVIDIA GPU arch:
    NVIDIA PTX archs:

  cuDNN:                         YES (ver 7.5.0)

  OpenCL:                        YES (NVD3D11)
    Include path:                D:/opencv_4/sources/3rdparty/include/opencl/1.2
    Link libraries:              Dynamic load

  Python 2:
    Interpreter:                 C:/ProgramData/Anaconda2/python.exe (ver 2.7.15)
    Libraries:                   C:/ProgramData/Anaconda2/libs/python27.lib (ver 2.7.15)
    numpy:                       C:/ProgramData/Anaconda2/lib/site-packages/numpy/core/include (ver 1.15.4)
    install path:                C:/ProgramData/Anaconda2/Lib/site-packages/cv2/python-2.7

  Python 3:
    Interpreter:                 C:/Users/starinfo/AppData/Local/Programs/Python/Python36/python3.exe (ver 3.6.8)
    Libraries:                   C:/Users/starinfo/AppData/Local/Programs/Python/Python36/libs/python36.lib (ver 3.6.8)
    numpy:                       C:/Users/starinfo/AppData/Roaming/Python/Python36/site-packages/numpy/core/include (ver 1.16.3)
    install path:                C:/Users/starinfo/AppData/Local/Programs/Python/Python36/Lib/site-packages/cv2/python-3.6

  Python (for build):            C:/ProgramData/Anaconda2/python.exe

  Java:                          
    ant:                         NO
    JNI:                         C:/Program Files/Java/jdk1.8.0_211/include C:/Program Files/Java/jdk1.8.0_211/include/win32 C:/Program Files/Java/jdk1.8.0_211/include
    Java wrappers:               NO
    Java tests:                  NO

  Install to:                    D:/opencv_4/build/install
-----------------------------------------------------------------

Configuring incomplete, errors occurred!
See also "D:/opencv_4/build/CMakeFiles/CMakeOutput.log".
See also "D:/opencv_4/build/CMakeFiles/CMakeError.log".

**Comment by YashasSamaga** [comment]
That is a bug which was fixed by #17571. This PR was merged into master few hours ago. You need the latest master.

**Comment by abdou31** [comment]
This is really very strange, can you tell me what should I use, did you mean that I should download OpenCV 4.3 from GitHub to build it instead of 4.4.0 to avoid this error?

**Comment by YashasSamaga** [comment]
@abdou31 A bug was introduced a week ago which broke CUDA builds. The fix was merged into master just a few hours ago. You need to pull the latest changes from master to avoid that error.

Or you can use OpenCV 4.3 which is a stable version without any serious issues.

**Comment by abdou31** [comment]
Thank you, I'm trying to fix the error

**Comment by dkurt** [comment]
Please do not use issues for questions. Can we close it?

**Comment by abdou31** [comment]
Yes , you can close it.

**Comment by Jackyhh** [comment]
> This is really very strange, can you tell me what should I use, did you mean that I should download OpenCV 4.3 from GitHub to build it instead of 4.4.0 to avoid this error?

Hello, I run successfully using opencv 4.3.0.38. (Note about environment: anaconda; install command: pip install opencv-python==4.3.0.38). However, it has a problem that my program can't use cuda backend, switching to CPU...

**Comment by YashasSamaga** [comment]
@Jackyhh You need to build from source to use the CUDA backend.
